introspection-engine/datamodel-renderer/src/datasource.rs

```diff
@@ -9,17 +9,27 @@ use crate::{Array, Commented, Env, Text, Value};
 #[derive(Debug)]
 pub struct Datasource<'a> {
     name: &'a str,
-    provider: Text<'a>,
+    provider: Text<&'a str>,
     url: Env<'a>,
     shadow_database_url: Option<Env<'a>>,
     relation_mode: Option<RelationMode>,
     custom_properties: Vec<(&'a str, Value<'a>)>,
     documentation: Option<Commented<'a>>,
-    namespaces: Array<'a>,
+    namespaces: Array<Text<&'a str>>,
 }
 
 impl<'a> Datasource<'a> {
     /// Create a new datasource with all required properties.
+    ///
+    /// ```ignore
+    /// datasource db {
+    /// //         ^^ name
+    ///   provider = "postgresql"
+    /// //            ^^^^^^^^^^ provider
+    ///   url      = env("DATABASE_URL")
+    /// //                ^^^^^^^^^^^^ url
```

Comment:
It's extreme nitpicking, but shouldn't the `env(` and `)` also be underlined?

Reply:
Yeah, that might be. I used that first... The string value the user gives us is going to be `DATABASE_URL`, so I'm pointing how it's rendered in the schema...

It's not super obvious what to do here, also not super important to decide one way or another.